### PR TITLE
Enables reclaim disk space feature.

### PR DESCRIPTION
**Pull request by ipanova**
[noissue]

### CODE REVIEW

File: pulp_container/app/registry.py [hunk 1]
```diff
@@ -163,7 +156,8 @@ async def dispatch_tag(tag, response_headers):
         try:
             artifact = tag.tagged_manifest._artifacts.get()
         except ObjectDoesNotExist:
-            raise ArtifactNotFound(tag.name)
+            ca = tag.tagged_manifest.contentartifact_set.all()[0]
```
Comment (ipanova) [inline pulp_container/app/registry.py:159]: this change is actually redundant if we 'protect' the manifests from being reclaimed.

File: pulp_container/app/models.py [hunk 2]
```diff
@@ -91,6 +93,8 @@ class Manifest(Content):
             field is used only for a manifest-list type Manifests.
     """
 
+    protected_from_reclaim = True
```
Comment (mdellweg) [inline pulp_container/app/models.py:96]: I would introduce this class constant defaulting to to True in pulpcore and only overwrite if reclaim is supported.
This way plugins can opt in to this feature, and don't need to opt out.
Also it's supposed to be a constant, so it should be all caps.

Comment (ipanova) [reply pulp_container/app/models.py:96]: it is set to true in core, it is just mentioned here in a redundant way so i will remove it

File: pulp_container/app/registry.py (hunk 1 above)
Comment (ipanova) [reply pulp_container/app/registry.py:159]: I have decided to enable reclaim for manifests as well, it required to tweak the logic in schema_conversion.

Comment (mdellweg) [reply pulp_container/app/registry.py:159]: What if the manifest is broken and has not artifact nor remote artifact?

**Review by mdellweg — APPROVED** [review]

File: pulp_container/app/registry.py (hunk 1 above)
Comment (ipanova) [reply pulp_container/app/registry.py:159]: actually handler handles this just fine - 404will be raised https://github.com/pulp/pulpcore/blob/master/pulpcore/content/handler.py#L681

File: CHANGES/9169.feature [hunk 3]
```diff
@@ -0,0 +1 @@
+Enable reclaim disk space feature for blobs and manifests.
```
Comment (mdellweg) [inline CHANGES/9169.feature:1]: We should note, it's only working with pulpcore 3.15.

**Pull request merged**